src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImplIntTest.kt

```diff
@@ -97,6 +103,73 @@ class BookingResourceImplIntTest : ResourceTest() {
       verify(inmateRepository).findPersonalCareNeeds(bookingId.toLong(), setOf("DISAB", "MATSTAT"))
     }
 
+    @Test
+    fun getAllpersonalCareNeeds() {
+      val bookingId = -1
+      val referenceCodesByDomain: List<ReferenceCode> = listOf(
+        ReferenceCode.builder().code("DISAB").domain("domain").description("Description 1").activeFlag("Y").build(),
+        ReferenceCode.builder().code("MATSTAT").domain("domain").description("Description 2").activeFlag("Y").build(),
+      )
+      `when`(referenceDomainService.getReferenceCodesByDomain("HEALTH_PBLM")).thenReturn(referenceCodesByDomain)
+      referenceCodesByDomain.stream().map<String?> { obj: ReferenceCode? -> obj!!.code }
```

Comment:
What is this line doing?

Reply:
Ah I see... it was getting a list of codes to pass into the mocking below.  Worth doing instead of using ArgumentMatchers?